src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -169,7 +169,7 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
         // it until the next tick.
         workInProgress.child = current.child;
         reuseChildren(workInProgress, workInProgress.child);
-        if (workInProgress.pendingWorkPriority <= priorityLevel) {
+        if (workInProgress.pendingWorkPriority !== NoWork && workInProgress.pendingWorkPriority <= priorityLevel) {
```

Comment:
Two options there:
- Allow negative priorities (or make every priority negative so that the ordering is maintained)
- Allow floating point number priorities

Not sure if they are better than the status quo though